pkg/controller/predicates/resources/resources.go

```diff
@@ -6,33 +6,36 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
-var _ predicate.Predicate = DeploymentPredicate{}
-
-type DeploymentPredicate struct {
-	predicate.Funcs
-}
-
-// Update implements default UpdateEvent filter for validating generation change.
-func (DeploymentPredicate) Update(e event.UpdateEvent) bool {
-	if e.ObjectOld == nil || e.ObjectNew == nil {
-		return false
-	}
-
-	oldDeployment, ok := e.ObjectOld.(*appsv1.Deployment)
-	if !ok {
-		return false
+func NewDeploymentPredicate() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return true
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return true
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return true
+		},
```

Comment:
I see that the default implementation of these returns true if they're not specified, so potentially could be omitted?

Reply:
done